### PR TITLE
PostgreSQL: bump lib/pq to latest version

### LINKED ISSUE #72417: PostgreSQL: Can't connect to SNI routed database
### What happened?

Grafana is using the `lib/pq` Go Postgres driver version `v0.10.6`.  When attempting to connect to a PostgreSQL back end that routes based of SNI the connection will fail.

```
INFO [07-26|22:56:48] Starting Grafana                         logger=settings version=10.1.0-pre commit=3dc60cd2d7 branch=main compiled=2023-07-26T17:08:36-05:00
INFO [07-26|22:56:48] Config loaded from                       logger=settings file=/home/nhudson/code/nhudson/grafana/conf/defaults.ini
INFO [07-26|22:56:48] Config overridden from command line      logger=settings arg="app_mode=development"
INFO [07-26|22:56:48] Target                                   logger=settings target=[all]
INFO [07-26|22:56:48] Path Home                                logger=settings path=/home/nhudson/code/nhudson/grafana
INFO [07-26|22:56:48] Path Data                                logger=settings path=/home/nhudson/code/nhudson/grafana/data
INFO [07-26|22:56:48] Path Logs                                logger=settings path=/home/nhudson/code/nhudson/grafana/data/log
INFO [07-26|22:56:48] Path Plugins                             logger=settings path=/home/nhudson/code/nhudson/grafana/data/plugins
INFO [07-26|22:56:48] Path Provisioning                        logger=settings path=/home/nhudson/code/nhudson/grafana/conf/provisioning
INFO [07-26|22:56:48] App mode development                     logger=settings
INFO [07-26|22:56:48] Connecting to DB                         logger=sqlstore dbtype=postgres
ERROR[07-26|22:56:48] alert migration failure: could not get migration log logger=migrator error="failed to check table existence: EOF"
```

### What did you expect to happen?

I would expect the connection to successfully connect to the backend Postgres instance.

### Did this work before?

Not that I am aware of

### How do we reproduce it?

Locally:
1. In your `default.ini` configure the database to connect to a Postgres instance that is behind an SNI route.

```
[database]
type = postgres
url = postgres://postgres:password@database-behind-sni-route.com:5432/grafana?sslmode=require
```
2. Start a local Grafana instance with `make run`

### Is the bug inside a dashboard panel?

_No response_

### Environment (with versions)?

Grafana: 9.3.6
OS: Ubuntu
Browser: Firefox

### Grafana platform?

Kubernetes

### Datasource(s)?

_No response_

### PR DESCRIPTION
**Pull request by nhudson**
**What is this feature?**

Update `lib/pq` to the latest version 

**Why do we need this feature?**

With the current version of `lib/pq`, connections to a PostgreSQL database that use SNI routing will fail.  

**Who is this feature for?**

This is for anyone who is attempting to connect Grafana to a PostgreSQL database backend which sits behind a router that is routing the traffic with SNI.

**Which issue(s) does this PR fix?**:

Fixes #72417

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by zserge — APPROVED** [review]

**Comment by mildwonkey** [comment]
Hey @nhudson - thank you for opening this, and sorry for the delay since @zserge approved it. If you can fix the `go.sum` conflicts I'll get this merged; you can also let me know if you'd rather I fix that - I can push a commit if you don't have time right now. Thanks again!

**Comment by nhudson** [comment]
> Hey @nhudson - thank you for opening this, and sorry for the delay since @zserge approved it. If you can fix the `go.sum` conflicts I'll get this merged; you can also let me know if you'd rather I fix that - I can push a commit if you don't have time right now. Thanks again!

Thanks for the heads up!  I can get to it tomorrow if that's good?

**Comment by nhudson** [comment]
@mildwonkey @zserge I have fixed the conflicts, but Drone is failing and I am not really sure how to fix it.  Do one you of mind looking at it for me?

File: go.mod [hunk 1]
```diff
@@ -288,6 +288,8 @@ require (
 	k8s.io/klog/v2 v2.90.1 // @grafana/grafana-app-platform-squad
 )
 
+require github.com/blang/semver/v4 v4.0.0
```
Comment (mildwonkey) [inline go.mod:291]: I'm not sure where this dependency is coming from - I recreated this PR locally and it's not showing up. Do you have any ideas, or are you able to remove it?

**Review by mildwonkey — CHANGES_REQUESTED** [review]
The only failure we need to address is the modowners, and removing the extraneous semver import in go.mod should take care of that.

**Review by mildwonkey — APPROVED** [review]

**Comment by nhudson** [comment]
@mildwonkey I don't know what the new failures are from.  Is this something that I should investigate?

**Comment by mildwonkey** [comment]
thanks for sticking with this @nhudson - I appreciate the extra hoops you had to jump through! You should be all good - I will follow up on that last failing check, but I'm reasonably optimistic that it's not a blocker 


(I'm technically positive it's not a blocker because it is not a required check, but I'm still going to double check 😁 )

**Comment by mildwonkey** [comment]
and we're confirmed (in internal slack) - thanks again @nhudson !

**Pull request merged**